improver_tests/cli/test_init.py

```diff
@@ -158,30 +158,41 @@ def test_without_output(self, m):
 
     @patch("improver.utilities.save.save_netcdf")
     def test_with_output(self, m):
-        """Tests that save_netcdf is called with object and string and default
-        compression_level=1"""
+        """Tests that save_netcdf is called with object and string, default
+        compression_level=1 and default least_significant_digit=None"""
         # pylint disable is needed as it can't see the wrappers output kwarg.
         # pylint: disable=E1123
         result = wrapped_with_output(2, output="foo")
-        m.assert_called_with(4, "foo", 1)
+        m.assert_called_with(4, "foo", 1, None)
         self.assertEqual(result, None)
 
     @patch("improver.utilities.save.save_netcdf")
     def test_with_output_compression_level(self, m):
-        """Tests that save_netcdf is called with object and string and compression_level=9"""
+        """Tests that save_netcdf is called with object and string, compression_level=9 and default least_significant_digit=None"""
         # pylint disable is needed as it can't see the wrappers output kwarg.
         # pylint: disable=E1123
         result = wrapped_with_output(2, output="foo", compression_level=9)
-        m.assert_called_with(4, "foo", 9)
+        m.assert_called_with(4, "foo", 9, None)
         self.assertEqual(result, None)
 
     @patch("improver.utilities.save.save_netcdf")
     def test_with_output_no_compression(self, m):
-        """Tests that save_netcdf is called with object and string and compression_level=0 """
+        """Tests that save_netcdf is called with object and string, compression_level=0 and default least_significant_digit=None"""
         # pylint disable is needed as it can't see the wrappers output kwarg.
         # pylint: disable=E1123
         result = wrapped_with_output(2, output="foo", compression_level=0)
-        m.assert_called_with(4, "foo", 0)
+        m.assert_called_with(4, "foo", 0, None)
+        self.assertEqual(result, None)
+
+    @patch("improver.utilities.save.save_netcdf")
+    def test_with_output_with_least_significant_figure(self, m):
+        """Tests that save_netcdf is called with object and string, compression_level=0 and least_significant_digit=2 """
+        # pylint disable is needed as it can't see the wrappers output kwarg.
+        # pylint: disable=E1123
+        result = wrapped_with_output(
+            2, output="foo", compression_level=0, least_significant_digit=2
+        )
```

Comment:
I think it could be better to decorate `wrapped_with_output` with `@clizefy` and use` wrapped_with_output.cli` in `Test_with_output`:

```python
result = wrapped_with_output.cli(
            "argv[0]", "2", "--output=foo", "--compression-level=0", "--least-significant-digit=2"
        )
```

This way we would exercise also the type conversion (if you do that, `Test_clizefy` should probably be moved up to run beforehand).

(PS: I would hope we port this file fully to `pytest` some day - `unittest.TestCase` style is not nice to look at).

Reply:
Okay, done - thanks!

Yeah I hope so, I think unittest is used in a few places atm, we should add replacing it to the "To-Do" list

Reply:
Thanks 👍